### DISCUSSION TITLE
struct types and reflection

### DISCUSSION THREAD

**Discussion by unicomp21** [esfx/esfx#52] [Q&A] [answered]
Is it possible to reflect across struct types?  My reason for asking, it would be a pretty awesome way to generate binary and json serialize/deserialize code across many/all languages?  Similar to protobuf and grpc?  But much simpler?

**Comment by rbuckton** [comment, marked as answer]
It would be feasible to surface the type information associated with a `struct`, but it isn't something I'm planning on doing currently.

**Reply by unicomp21** [reply]
Crossing my fingers, and hoping @rbuckton will get bored, lol.  And build us and rpc framework, lol.

Btw, the connect-go stuff is looking interesting.
https://github.com/bufbuild/connect-go
